library/src/device/generator/generator.kernel.hpp

```diff
@@ -342,21 +347,25 @@ namespace StockhamGenerator
             /*=======================================================*/
             std::string loop;
             loop += "\tif(dim == 1){\n";
-            loop += "\t\t" + offset_name + " += counter_mod*" + stride_name + "[1];\n";
+            loop += "\t\t" + offset_name1 + " += counter_mod*" + stride_name1 + "[1];\n";
+            if (output == true) loop += "\t\t" + offset_name2 + " += counter_mod*" + stride_name2 + "[1];\n";
```

Comment:
I get that, but doesn't line 350 end the generated line with a ";".  And then if output is true, we will have a +... following the ";" in the generated code, which is legal, but doesn't actually do anything.